src/tests/state/router/selectors.test.js

```diff
@@ -0,0 +1,156 @@
+import { LOCATION_CHANGE } from 'connected-react-router/lib/actions';
+import { fromJS } from 'immutable';
+import * as matchers from 'jest-immutable-matchers';
+
+import {
+  getCurrentPathname,
+  getCurrentRoute,
+  getCurrentRouteTree,
+  getCurrentUrlAsObject,
+  getQueryParam,
+  getQueryParamAsInt,
+  getUrlParam,
+  getUrlParamAsInt,
+} from 'console/state/router/selectors';
+import { rootReducer } from 'console/tests/mockStore';
+import { reduceUrlsToRoutesList } from 'console/utils/router';
+import * as consoleUrls from 'console/urls';
+
+const originalConsoleUrls = consoleUrls.default;
+
+let STATE = rootReducer(undefined, {
+  type: LOCATION_CHANGE,
+  payload: {
+    location: {
+      pathname: '/test/123/',
+      search: '?status=456',
+      hash: '',
+      key: '',
+    },
+    action: 'PUSH',
+  },
+});
+
+beforeAll(() => {
+  // Mock URL routing
+  consoleUrls.default = reduceUrlsToRoutesList({
+    '/': {
+      name: 'home',
+      component: null,
+      routes: {
+        '/test': {
+          name: 'test-listing',
+          component: null,
+          routes: {
+            '/:testId': {
+              name: 'test-details',
+              component: null,
+            },
+          },
+        },
+      },
+    },
+  });
+});
+
+afterAll(() => {
+  // Clean up mocked URLs
+  consoleUrls.default = originalConsoleUrls;
+});
+
+describe('getCurrentPathname', () => {
+  it('should return the current pathname', () => {
+    expect(getCurrentPathname(STATE)).toEqual('/test/123/');
+  });
+});
+
+describe('getCurrentRoute', () => {
+  it('should return the current route', () => {
+    const route = getCurrentRoute(STATE);
+    expect(route).toEqual({
+      component: null,
+      exact: true,
+      name: 'test-details',
+      parentPath: '/test/',
+      path: '/test/:testId/',
+    });
+  });
+});
+
+describe('getCurrentRouteTree', () => {
+  beforeEach(() => {
+    jest.addMatchers(matchers);
```

Comment:
I know it's copy/paste, but I think that adding this in the `beforeAll` function above makes more sense, instead of doing it before every single test.